starter_code/src/movies.js

```diff
@@ -1,15 +1,108 @@
+const films = require("./data")
+
 /* eslint no-restricted-globals: 'off' */
 
 // Iteration 1: Ordering by year - Order by year, ascending (in growing order)
+function orderByYear(arr) {
+    if (arr.length ===1) {
+        return arr
+    }
+    var newArray = [...arr];
+    let sortedMovies = newArray.sort(function(a,b) {
+        if (a.year !== b.year) {
+            return a.year - b.year;
+        } else return a.title.localeCompare(b.title)
+    }) 
+    return sortedMovies
+}
 
 // Iteration 2: Steven Spielberg. The best? - How many drama movies did STEVEN SPIELBERG direct
 
+function howManyMovies(arr) {
+    var counter = 0;
+    arr.filter(function(a) {
+        if (a.director === "Steven Spielberg" && a.genre.includes("Drama")){
+            counter++
+        }
+    }); return counter
+}
+
 // Iteration 3: Alphabetic Order - Order by title and print the first 20 titles
 
+function orderAlphabetically(arr) {
+    var newArray = [];
+    if (arr.length < 20) {
+        newArray = arr.map(function(a){
+            return a.title;
+        })
+    } if (arr.length > 20) {
+        newArray = arr.splice(0, 20).map(function(a){
+            return a.title
+        }).sort(function(a,b){
+           return a.localeCompare(b)
+        })
+    }; return newArray;
+}
```

Comment:
It's not accepted by Jasmine because it's not working when the array has less than 20 elements :D.
You still have to  sort the array even if there are less than 20 elements.